core/src/main/java/com/novoda/merlin/service/MerlinServiceBinder.java

```diff
@@ -70,16 +69,31 @@ private boolean connectionIsAvailable() {
         @Override
         public void onServiceConnected(ComponentName name, IBinder binder) {
             Logger.d("onServiceConnected");
-            merlinService = ((MerlinService.LocalBinder) binder).getService();
-            merlinService.setConnectListener(listenerHolder.connectListener);
-            merlinService.setDisconnectListener(listenerHolder.disconnectListener);
-            merlinService.setBindStatusListener(listenerHolder.bindListener);
-            merlinService.setHostname(endpoint, validator);
+            MerlinService.LocalBinder merlinServiceBinder = ((MerlinService.LocalBinder) binder);
```

Comment:
Create all `MerlinService` dependencies and pass through the minimum objects to `MerlinService`.